diff --git a/holoscan-sdk/Dockerfile b/holoscan-sdk/Dockerfile
index 57079df..0fac0a7 100644
--- a/holoscan-sdk/Dockerfile
+++ b/holoscan-sdk/Dockerfile
@@ -31,7 +31,7 @@ ARG GXF_VERSION=23.05_20230717_d105fa1c
 # Base image
 ############################################################
 ARG GPU_TYPE=dgpu
-FROM nvcr.io/nvidia/tensorrt:22.03-py3 AS dgpu_base
+FROM nvcr.io/nvidia/tensorrt:22.12-py3 AS dgpu_base
 FROM nvcr.io/nvidia/l4t-tensorrt:r8.5.2.2-devel AS igpu_base
 FROM ${GPU_TYPE}_base AS base
 
@@ -50,7 +50,7 @@ RUN rm -r \
     /usr/local/bin/cmake \
     /usr/local/bin/cpack \
     /usr/local/bin/ctest \
-    /usr/local/share/cmake-3.14
+    /usr/local/share/cmake-3.24
 RUN curl -s -L https://apt.kitware.com/keys/kitware-archive-latest.asc 2>/dev/null \
         | gpg --dearmor - \
         | tee /usr/share/keyrings/kitware-archive-keyring.gpg >/dev/null \
